README.md

```diff
@@ -1,58 +1,83 @@
 # ReevooLogger
 
-Welcome to your new gem! In this directory, you'll find the files you need to be able to package up your Ruby library into a gem. Put your Ruby code in the file `lib/reevoo_logger`. To experiment with that code, run `bin/console` for an interactive prompt.
-
-TODO: Delete this and the text above, and describe your gem
+ReevooLogger consolidates logstasher, statsd and sentry raven into a single library with one setup. It automatically
+tracks all log messages into logstash, exceptions into sentry and updates exceptions stats in statsd. It also allow you
+to call methods directly on instances of statsd and raven if needed.
 
 ## Installation
 
-### In your Gemfile: 
+### In your Gemfile:
 
 ```ruby
 gem 'reevoo_logger'
 ```
-    
+
 ### Init logger:
-    
-    module TestApp
-      def self.logger
-        @logger ||= ReevooLogger.new('app_name', Rack::Directory.new("").root)
-      end
-    end
-    
-You can specify the log output
 
-    @logger ||= ReevooLogger.new('app_name', Rack::Directory.new("").root, STDOUT)
-  
+```ruby
+module TestApp
+  def self.logger
+    @logger ||= ReevooLogger.new(
+      app_name: "foo_app",
+      root_dir: Rack::Directory.new("").root
+      device: STDOUT, # default is file log in log/logstasher.log
+      raven_conf: {
+        dns: "https://00c73aa8f93f4hbwjehb4r20af10afb@app.getsentry.com/502146" # public sentry DNS
+      },
+      statsd_conf: { # in most cases you should be fine with default localhost:8125
+        host: "my-host",
+        port: 1234
+      }
+    )
+  end
+end
+```
+
 
 ### Setup Grape request/exception logging
 
 Add to Gemfile
 
-    gem 'grape_logging'
-    
+```ruby
+gem 'grape_logging'
+```
+
 Setup in Grape::API class
-     
-    module TestApp
-      class API < Grape::API
-        logger TestApp.logger
-        use GrapeLogging::Middleware::RequestLogger, logger: TestApp.logger
-        
-        rescue_from TestApp::NotFound do |err|
-          # Tag your exception 
-          API.logger.info(exception: err, tags: "rescued_exception", status: 404)
-          error_response(message: "Not found", status: 404)
-        end
-
-        rescue_from :all do |e|
-          API.logger.error(e)
-          error_response(message: e.message, status: 500)
-        end
-      end
+
+```ruby
+module TestApp
+  class API < Grape::API
+    logger TestApp.logger
+    use GrapeLogging::Middleware::RequestLogger, logger: TestApp.logger
+
+    rescue_from TestApp::NotFound do |err|
+      # Tag your exception
+      API.logger.info(exception: err, tags: "rescued_exception", status: 404)
+      error_response(message: "Not found", status: 404)
     end
 
+    rescue_from :all do |e|
+      API.logger.error(e)
+      error_response(message: e.message, status: 500)
+    end
+  end
+end
+```
+
+
+### Using statsd directly
+
+```ruby
+  TestApp.logger.statsd.increment('foo.bar')
+```
+
+### Using raven directly
+
+```ruby
+  TestApp.logger.raven.capture_exception(exception)
```

Comment:
I would prefer if we insulate the caller from the fact that they are using raven. Perhaps we want to use some other tool in the future, it would be nice to keep this gem compatible and just switch the backend out.

e.g.
```
TestApp.logger.capture_exception(exception)
# or
TestApp.logger.exceptions.capture(exception)
```

We should just wrap the subset of raven we think is important.

Also could we support the syntax where you wrap a call that may explode in a block

e.g.

```
TestApp.logger.capture_exception do
  Firework.explode
end
```